Microsoft.Toolkit.Uwp.UI/ImageCache/ImageCache.cs

```diff
@@ -24,11 +26,29 @@ namespace Microsoft.Toolkit.Uwp.UI
     /// </summary>
     public static class ImageCache
     {
+        /// <summary>
+        /// Helper class used to store BitmapImages to in-memory cache
+        /// </summary>
+        protected class MemoryCacheItem
+        {
+            /// <summary>
+            /// Gets or sets dateTime when image was last stored to in-memory cache
+            /// </summary>
+            public DateTime LastUpdated { get; set; }
+
+            /// <summary>
+            /// Gets or sets BitmapImage of in-memory cached item
+            /// </summary>
+            public BitmapImage Image { get; set; }
+        }
+
         private const string CacheFolderName = "ImageCache";
 
         private static readonly SemaphoreSlim _cacheFolderSemaphore = new SemaphoreSlim(1);
-        private static readonly Dictionary<string, Task> _concurrentTasks = new Dictionary<string, Task>();
+        private static readonly Dictionary<string, Task<BitmapImage>> _concurrentTasks = new Dictionary<string, Task<BitmapImage>>();
         private static StorageFolder _cacheFolder;
+        private static OrderedDictionary _memoryCache = new OrderedDictionary();
```

Comment:
I'm not sure if OrderedDictionary is the right choice for memory cache. It has to be ordered so it can be of a fixed size.